packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,687 @@
+
+*file toolbar-filter-state.js:*
+```jsx
+...
+import {
+    ..., Action
+} from '@devexpress/dx-react-core';
+
+export class ToolbarFilterState extends React.PureComponent {
+  constructor(props) {
+    super(props);
+        ...
+        this.state = {
+            filterValue: this.props.defaultFilterValue
+        };
+        this.changeValue = this.changeValue.bind(this);
+  }
+
+  changeValue() {
+  }
+
+  render() {
+    return (
+      <Plugin
+        name="ToolbarFilterState"
+      >
+        <Action name="changeToolbarFilterValue" action={this.changeValue} />
+      </Plugin>
+    );
+  }
+}
+```
+Now, `changeToolbarFilterValue` is available in any plugin that is used with `ToolbarFilterState`. In our scenario, we may get it using [TemplateConnector](../../../core/docs/reference/template-connector.md) in the `ToolbarFilter` plugin. The following code snippet shows how to get the `changeToolbarFilterValue` action and use it as the `Select.onChange` event handler.
+
+*toolbar-filter.js:*
+```jsx
+...
+export class ToolbarFilter extends React.PureComponent {
+  render() {
+    return (
+      <Plugin
+        ...
+      >
+        <Template name="toolbarContent">
+          <TemplatePlaceholder />
+          <TemplateConnector>
+          {({
+            ... //Accessing getters
+           }, {
+            changeToolbarFilterValue //Accessing actions
+          }) => (
+            <Select
+              ...
+              onChange={changeToolbarFilterValue}
+            >
+              ...
+            </Select>
+          )}
+          </TemplateConnector>
+        </Template>
+      </Plugin>
+    );
+  }
+}
+```
+
+The event handler passes the `event` parameter. We can get a new value using this parameter and pass it to the state object of the `ToolbarFilterState` plugin.
+
+*file toolbar-filter-state.js:*
+```jsx
+...
+export class ToolbarFilterState extends React.PureComponent {
+  ...
+  changeValue(event) {
+    this.setState({
+      filterValue: event.target.value
+    });
+  }
+  ...
+}
+```
+
+We managed to detect changes in the select box. However, it is not sufficient for filtering the grid. We will need to add the [IntegratedFiltering](../reference/integrated-filtering.md) plugin that performs built-in data filtering. If we take a look at the plugin description, we will see that there is the `filterExpression` getter that represents a filter expression that will be used for filtering. So, we will redefine the value of `filterExpression` using a computed function to extend the expression with our selected value.
+
+*index.js:*
+```jsx
+...
+import {
+  IntegratedFiltering
+} from '@devexpress/dx-react-grid';
+
+class App extends Component {
+  ...
+  render() {
+    ...
+    return (
+      <Paper>
+        <Grid
+          ...
+        >
+          <ToolbarFilterState defaultColumnName={"CompanyName"} defaultFilterValue={""}/>
```

Comment:
These curly braces can be removed since we pass string literals: 
`defaultColumnName="CompanyName"`